vassal-app/src/main/java/VASSAL/configure/VisibilityCondition.java

```diff
@@ -20,4 +20,19 @@
 @FunctionalInterface
 public interface VisibilityCondition {
   boolean shouldBeVisible();
+
+  class VisibilityAND implements VisibilityCondition {
+    VisibilityCondition and1;
+    VisibilityCondition and2;
+
+    public VisibilityAND(VisibilityCondition a1, VisibilityCondition a2) {
+      and1 = a1;
+      and2 = a2;
+    }
+
+    @Override
+    public boolean shouldBeVisible() {
+      return ((and1 == null) || and1.shouldBeVisible()) && ((and2 == null) || and2.shouldBeVisible());
+    }
+  }
```

Comment:
`VisibilityAND` should probably be in its own file, and have `final` members.

Reply:
fixed